### PR TITLE
feat: Attempt to add golang support. Still need to find out why

### PR DESCRIPTION
**Pull request by teddylear**
Attempt to add golang support. Still need to find out why parameters from parent function are not being added to extracted function. Treesitter query appears to be correct when checking a go project with Treesitter playground. Let me know if you want me to look into resolving that or if I could help with anything else (testing etc). New to lua but love the idea of this project!

### CODE REVIEW

File: lua/refactoring/config.lua [hunk 1]
```diff
@@ -6,6 +6,9 @@ local config = {
         lua = {
             -- cmd = [[ !stylua % ]],
         },
+        go = {
+
```
Comment (ThePrimeagen) [inline lua/refactoring/config.lua:10]: perhaps gofmt could be used here.  I assume its the "defacto" format lib.  You could add a quick check before executing it so it doesn't vomit errors on all golangers.

File: lua/refactoring/config.lua [hunk 2]
```diff
@@ -62,6 +65,32 @@ end
                 }
             end,
         },
+        go = {
+            extract_function = function(opts)
+                return {
+                    create = string.format(
+                        [[
+func %s(%s) {
```
Comment (ThePrimeagen) [inline lua/refactoring/config.lua:73]: I AM SO SORRY FOR THIS

Going to figure out a better way.

File: queries/go/refactoring.scm [hunk 3]
```diff
@@ -0,0 +1,16 @@
+;; Grabs all the local variable declarations.  This is useful for scope
+;; variable passing.  Which variables do we need to pass to the extracted
+;; function?
+((short_var_declaration
```
Comment (ThePrimeagen) [inline queries/go/refactoring.scm:4]: Let me try these out to make sure its correct.

**Review by ThePrimeagen — APPROVED** [review]

**Comment by ThePrimeagen** [comment]
As of now nothing stands out here.  I didn't look at the code generation closely, will circle back to it.

The formatting is the only thing I would like to have figured out in the near term, but first let me see the queries you are running.

File: queries/go/refactoring.scm (hunk 3 above)
Comment (teddylear) [reply queries/go/refactoring.scm:4]: Added tests to make validation of this easier. Thanks again for the test harness! Made fixing these so much easier.

File: lua/refactoring/config.lua (hunk 1 above)
Comment (teddylear) [reply lua/refactoring/config.lua]: Tried to add gofmt, but ran into similar issue I assumed you ran into with stylua where it's vomiting on the `%` while testing. Tried to run manually but it still wasn't formatting either... Let me know if you want me to open another issue to look into this as well.

**Comment by teddylear** [comment]
Added the tests. Only major change I had to make was to break out the `command` files used per test into their own files as I'm assuming it will change per language (and eventually other refactoring commands). Let me know if you want me to do another approach of resolving the command per test issue.

File: lua/refactoring/config.lua (hunk 1 above)
Comment (ThePrimeagen) [reply lua/refactoring/config.lua]: yeah, it has to do with the fact that you are editing the file, with changes, outside of vim.  Causing two different file trees.

File: queries/go/refactoring.scm (hunk 3 above)
Comment (ThePrimeagen) [reply queries/go/refactoring.scm:4]: awsome!

**Comment by ThePrimeagen** [comment]
Hurts a little that commands had to be broken out per language, but it also makes sense.  Not a lot we can do about it :(

**Comment by ThePrimeagen** [comment]
We have a PR that is editing the config file that will cause one of you two to get a merge conflict.  Do you wish to die by the sword or the other? :)

**Comment by ThePrimeagen** [comment]
LGTM.  Sorry for #22 .  We just created a conflict :(

**Comment by teddylear** [comment]
I can update my PR tomorrow, feel free to merge the other PR first

**Pull request merged**